### PR TITLE
EDM-1800: agent: implement new poll backoff for podman and queue 

### PR DESCRIPTION
**Pull request by hexfusion**
This PR introduces a new pool package with the mid-term goal of eliminating the wait dependency from API machinery.
It also integrates this pool into the podman package and improves resync poll durations by adding incremental backoff logic.

```
Jul 03 13:08:35 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:35.652265Z" level=debug msg="Requeueing spec" file="agent/device/device.go:167"
Jul 03 13:08:35 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:35.652385Z" level=debug msg="Completed sync of device spec in 2.045987ms" file="agent/device/device.go:158"
Jul 03 13:08:35 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:35.652502Z" level=debug msg="Started collecting device status" file="agent/device/device.go:288"
Jul 03 13:08:35 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:35.657919Z" level=debug msg="No new template version from management service" file="device/publisher/publisher.go:124"
Jul 03 13:08:35 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:35.732994Z" level=debug msg="Completed pushing device status in: 80.487663ms" file="agent/device/device.go:291"
Jul 03 13:08:37 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:37.650585Z" level=debug msg="Starting sync of device spec" file="agent/device/device.go:155"
Jul 03 13:08:37 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:37.651290Z" level=debug msg="Requeuing current desired spec from disk version: 1" file="device/spec/manager.go:353"
Jul 03 13:08:37 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:37.651396Z" level=trace msg="Skipping item with version 1 already in queue" file="device/spec/queue.go:294"
Jul 03 13:08:37 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:37.651462Z" level=trace msg="Popped item version 1, heap size now 0" file="device/spec/queue.go:317"
Jul 03 13:08:37 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:37.651553Z" level=debug msg="Evaluating template version: 1" file="device/spec/queue.go:116"
Jul 03 13:08:37 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:37.651640Z" level=trace msg="Added item version 1, heap size now 1" file="device/spec/queue.go:308"
Jul 03 13:08:37 localhost.localdomain flightctl-agent[1696]: time="2025-07-03T13:08:37.651698Z" level=debug msg="Template version 1 requeue is currently in backoff. Available after: 2025-07-03T13:09:42Z" file="device/spec/queue.go:121"
```

## Summary by CodeRabbit

* **New Features**
  * Introduced a new exponential backoff polling mechanism, providing more flexible and configurable retry behavior for operations.
  * Added a unified polling configuration for application and spec management, replacing fixed delay thresholds.
  * Added a new polling configuration utility for test environments.
  * Added a maximum retry timeout for Podman client operations.

* **Bug Fixes**
  * Improved reliability of retry logic by capping maximum delays and supporting context-based cancellation.
  * Enhanced retryable error detection to include maximum retry step errors.

* **Tests**
  * Updated and expanded tests to validate new backoff and polling behavior, including edge cases and timing scenarios.

* **Chores**
  * Standardized polling configuration across tests for consistency.
  * Removed deprecated retry delay parameters and replaced them with unified poll configurations.

### CODE REVIEW

File: internal/agent/client/podman.go [hunk 1]
```diff
@@ -596,9 +596,9 @@ func SanitizePodmanLabel(name string) string {
 	return result.String()
 }
 
-func retryWithBackoff(ctx context.Context, log *log.PrefixLogger, backoff wait.Backoff, operation func(context.Context) (string, error)) (string, error) {
+func retryWithBackoff(ctx context.Context, log *log.PrefixLogger, backoff poll.Config, operation func(context.Context) (string, error)) (string, error) {
```
Comment (kkyrazis) [inline internal/agent/client/podman.go:600]: I could be misunderstanding this, but the new poll package appears to perform retries up until a total duration is reached (`defaultPodmanTimeout`).  A call to something like `Pull` defers to the `pull` with a timeout of `defaultPodmanTimeout`. This could potentially lead to only a single pull attempt could be made within a `retryWithBackOff`. Previously, the retry would perform N steps without regards to total timeout. Is this new behavior intentional?

Comment (hexfusion) [reply internal/agent/client/podman.go:600]: aha good catch yes I am using this PR as a stepping stone BUT the goal was not to change the flow here. saying that the total backoff should be about an hour with each try timing out at 10m.

Comment (hexfusion) [reply internal/agent/client/podman.go:600]: I also added MaxSteps to align with wait.

File: internal/agent/client/compose.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ const (
 	ComposeOverrideFilename      = "99-compose-flightctl-agent.override.yaml"
 	ComposeDockerProjectLabelKey = "com.docker.compose.project"
 	defaultPodmanTimeout         = 10 * time.Minute
+	defaultPodmanMaxRetryTimeout = 60 * time.Minute
```
Comment (hexfusion) [inline internal/agent/client/compose.go:28]: note: this is just the global max timeout. the value is the same as before if we consider 6 steps at 10m each.

File: pkg/poll/poll.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		return fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	delay := cfg.BaseDelay
+	attempts := 0
+	for {
+		if cfg.MaxSteps > 0 && attempts >= cfg.MaxSteps {
```
Comment (kkyrazis) [inline pkg/poll/poll.go]: I think this check would work better after `attempts++` otherwise we will delay for the next attempt that will never occur.

Comment (hexfusion) [reply pkg/poll/poll.go]: great point why wait. done

**Review by kkyrazis — APPROVED** [review]
